.travis.yml

```diff
@@ -112,6 +115,21 @@ script:
       python -c 'import mpi4py' &&
       mpiexec -np 4 python pymultinest_demo.py;
     fi
+  - if [[ "$WITHMPI" == "no" ]]; then
+      if [[ "$MULTINEST_TAG" == "filelength" ]]; then
+        python pymultinest_demo_longname.py;
+      else
+        TESTRESULT=0; python pymultinest_demo_longname.py || TESTRESULT=$?; expr $TESTRESULT != 0;
```

Comment:
Yes, when the test python script will fail due to the line length being too long and return an exit code value of 2. This would automatically cause travis to say that the test has failed. However, as we expect it to fail we can catch the output in `TESTRESULT` using the or `||` expression - the whole `(... || ...)` expression will return 0 so travis will be happy, but we can then check the scripts exit code a make sure it didn't return a 0. Credit for this, as usual, goes to StackOverflow!

Reply:
Perhaps a 
```
if python pymultinest_demo_longname.py
then 
    echo "unexpected success"; exit 1;
else
    echo "long name failed with error code $?, as expected"
fi
```
would be clearer.

Reply:
Unfortunately this doesn't work. I don't think `exit` can be used outside of a script to return an exit code.

Reply:
Actually, it does work. I just had a different typo that was causing the problem!